src/WebJobs.Script/Extensions/HttpRequestMessageExtensions.cs

```diff
@@ -15,6 +16,19 @@ public static AuthorizationLevel GetAuthorizationLevel(this HttpRequestMessage r
             return request.GetRequestPropertyOrDefault<AuthorizationLevel>(ScriptConstants.AzureFunctionsHttpRequestAuthorizationLevel);
         }
 
+        public static bool IsAntaresInternalRequest(this HttpRequestMessage request)
+        {
+            if (!ScriptSettingsManager.Instance.IsAzureEnvironment)
+            {
+                return false;
+            }
+
+            // this header will *always* be present on requests originating externally (i.e. going
+            // through the Anatares front end). For requests originating internally it will NOT be
+            // present.
+            return !request.Headers.Contains(ScriptConstants.AntaresInternalRequestHeaderName);
```

Comment:
The naming of `AntaresInternalRequestHeaderName` is confusing, I think it should be `ExternalRequestHeaderName` or something along those lines, based on how it is used here